File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    
    heap = []
    for i in scoville:
        heapq.heappush(heap,i)


    num = heapq.heappop(heap)
    while num < K:

        if len(heap) >= 2 :
            heapq.heappush(heap, num + heapq.heappop(heap)*2)
            answer += 1    
        else:
            if len(heap) == 1:
                if num + heapq.heappop(heap)*2 < K:
                    answer = -1
                else:
                    answer += 1
                break
            else:
                answer = -1
                break
        num = heapq.heappop(heap)
    
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        Arrays.sort(scoville);
        
        PriorityQueue<Integer> arr = new PriorityQueue<>();
        for(int i : scoville) arr.add(i);
        
        while(arr.peek() < K) {
            if (arr.size() < 2) {
                answer = -1;
                break;
            }
            int a = arr.poll();
            int b = arr.poll();
            arr.add(a+(b*2));
            answer += 1;
        }
        
        return answer;
    }
}